git-multimail/git_multimail.py

```diff
@@ -3756,11 +3759,13 @@ def run_as_post_receive_hook(environment, mailer):
         changes.append(
             ReferenceChange.create(environment, oldrev, newrev, refname)
             )
-    if changes:
-        push = Push(environment, changes)
+    if not changes:
+        return
```

Comment:
When following this `return`, the method exits without closing the mailer. The connection is open in the constructor of `SMTPMailer`, not in the `send` method, so even if no mail is sent we still need to properly close the connection.

I don't think all this code change is needed. It should be sufficient to rename `__init__` to `close`, to re-add an `__init__` method that just calls `close`, and to change the call to `__init__` to `close`.

Reply:
Yeah, some of this was just to match the other hook's structure more, but I can change it.

Reply:
Hopefully the addition of call to close from  __del__ covers all those code paths now.

Reply:
> Hopefully the addition of call to close from `__del__` covers all those code paths now.

I don't think it does: `mailer` is passed as an argument to the function, so a `return` does not call the destructor.

Reply:
Ahhh, that makes sense. I've never really used Python destructors before. Updating.